Index: test_mathematical_operations.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test_mathematical_operations.py b/test_mathematical_operations.py
new file mode 100644
--- /dev/null	(date 1712596260057)
+++ b/test_mathematical_operations.py	(date 1712596260057)
@@ -0,0 +1,11 @@
+from mathematical_operations import add
+from mathematical_operations import subtract
+from mathematical_operations import multiply
+from mathematical_operations import divide
+
+def test_mathematical_operations():
+    assert add(1,2 ) == 3
+    assert subtract(1, 2) == -1
+    assert multiply(1, 2) == 2
+    assert divide(1, 2) == 0.5
+    assert divide(1, 0) == "Division by zero is not allowed!"
\ No newline at end of file
Index: mathematical_operations.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mathematical_operations.py b/mathematical_operations.py
new file mode 100644
--- /dev/null	(date 1712596451660)
+++ b/mathematical_operations.py	(date 1712596451660)
@@ -0,0 +1,34 @@
+def add(a, b):
+    """adds two numbers"""
+    return a + b
+
+def subtract(a, b):
+    """subtracts two numbers"""
+    return a - b
+
+def multiply(a, b):
+    """multiplies two numbers"""
+    return a * b
+
+def divide(a, b):
+    """divides two numbers"""
+    if b != 0:
+        return a / b
+    else:
+        return "Division by zero is not allowed!"
+
+# data input
+num1 = float(input("Enter first number: "))
+num2 = float(input("Enter second number: "))
+
+# calculations
+result_add = add(num1, num2)
+result_subtract = subtract(num1, num2)
+result_multiply = multiply(num1, num2)
+result_divide = divide(num1, num2)
+
+# displays results2
+print("Addition:", result_add)
+print("Subtraction:", result_subtract)
+print("Multiplication:", result_multiply)
+print("Division:", result_divide)
